flang/lib/Optimizer/Dialect/FIROps.cpp

```diff
@@ -1273,6 +1273,85 @@ mlir::OpFoldResult fir::MulfOp::fold(llvm::ArrayRef<mlir::Attribute> opnds) {
       opnds, [](APFloat a, APFloat b) { return a * b; });
 }
 
+//===----------------------------------------------------------------------===//
+// ReboxOp
+//===----------------------------------------------------------------------===//
+
+/// Get the rank and element type of a box type.
+/// Example: return [2, f32] for !fir.box<!fir.heap<!fir.array<?x?xf32>>.
+static std::pair<unsigned, mlir::Type>
+getEleTyAndRankOfBoxType(mlir::Type boxTy) {
```

Comment:
Why not two separate functions? In fact, the rank is trivially found from a sequence type.

Reply:
I though it was less code, but I can understand its interface is more complex than two functions. I have updated the patch with two separate function instead.